modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/Async.kt

```diff
@@ -22,227 +22,236 @@ typealias Proc<A> = ((Either<Throwable, A>) -> Unit) -> Unit
+     *  //sampleEnd
+     *  println(result)
+     * }
+     * ```
+     *
+     * @see async for a simpler, non suspending version.
+     */
+    fun <A> asyncF(k: ProcF<F, A>): Kind<F, A>
+
+    /**
+     * Continue the evaluation on provided [CoroutineContext]
+     *
+     * @param ctx [CoroutineContext] to run evaluation on
+     *
+     * ```kotlin:ank:playground:extension
+     * _imports_
+     * import kotlinx.coroutines.Dispatchers
+     *
+     * fun main(args: Array<String>) {
+     *   //sampleStart
+     *   fun <F> Async<F>.runOnDefaultDispatcher(): Kind<F, String> =
+     *     _just_(Unit)._continueOn_(Dispatchers.Default).flatMap {
+     *       _delay_({ Thread.currentThread().name })
+     *     }
+     *
+     *   val result = _extensionFactory_.runOnDefaultDispatcher()
+     *   //sampleEnd
+     *   println(result)
+     * }
+     * ```
+     */
+    fun <A> Kind<F, A>.continueOn(ctx: CoroutineContext): Kind<F, A>
+
+    /**
+     * Delay a computation on provided [CoroutineContext].
+     *
+     * @param ctx [CoroutineContext] to run evaluation on.
+     *
+     * ```kotlin:ank:playground:extension
+     * _imports_
+     * import kotlinx.coroutines.Dispatchers
+     *
+     * fun main(args: Array<String>) {
+     *   //sampleStart
+     *   fun <F> Async<F>.invokeOnDefaultDispatcher(): Kind<F, String> =
+     *     _delay_(Dispatchers.Default, { Thread.currentThread().name })
+     *
+     *   val result = _extensionFactory_.invokeOnDefaultDispatcher()
+     *   //sampleEnd
+     *   println(result)
+     * }
+     * ```
+     */
+    fun <A> delay(ctx: CoroutineContext, f: () -> A): Kind<F, A> =
+            defer(ctx) {
+                try {
+                    just(f())
+                } catch (t: Throwable) {
+                    t.raiseNonFatal<A>()
+                }
+            }
+
+    @Deprecated("Use delay instead",
+            ReplaceWith("delay(ctx, f)", "arrow.effects.typeclasses.Async"))
+    operator fun <A> invoke(ctx: CoroutineContext, f: () -> A): Kind<F, A> =
+            delay(ctx, f)
+
+    /**
+     * Delay a computation on provided [CoroutineContext].
+     *
+     * @param ctx [CoroutineContext] to run evaluation on.
+     *
+     * ```kotlin:ank:playground:extension
+     * _imports_
+     * import kotlinx.coroutines.Dispatchers
+     *
+     * fun main(args: Array<String>) {
+     *   //sampleStart
+     *   fun <F> Async<F>.invokeOnDefaultDispatcher(): Kind<F, String> =
+     *     _defer_(Dispatchers.Default, { delay { Thread.currentThread().name } })
+     *
+     *   val result = _extensionFactory_.invokeOnDefaultDispatcher().fix().unsafeRunSync()
+     *   //sampleEnd
+     *   println(result)
+     * }
+     * ```
+     */
+    fun <A> defer(ctx: CoroutineContext, f: () -> Kind<F, A>): Kind<F, A> =
+            just(Unit).continueOn(ctx).flatMap { defer(f) }
+
+    /**
+     * Delay a computation on provided [CoroutineContext].
```

Comment:
`Delay a branching computation on the provided [CoroutineContext].`

And you could use this opportunity to add a doc on the original `delayOrRaise` too as  `Delay a branching computation` :D